ochopod/watchers/remote.py

```diff
@@ -80,12 +81,39 @@ def spin(self, data):
             #
             pods = {}
             try:
-                path = '%s/%s.%s/snapshot' % (ROOT, self.scope, self.tag)
-                value, stat = self.zk.get(path, watch=self.feedback)
-                try:
-                    pods = json.loads(value)
-                except ValueError:
-                    pass
+                # 
+                # - if no wildcard character is used, let us simply watch the zookeeper 
+                #   node we are interested in, which is best in terms of performance
+                #
+                if "*" not in self.tag:
+                    path = '%s/%s.%s/snapshot' % (ROOT, self.scope, self.tag)
+                    value, stat = self.zk.get(path, watch=self.feedback)
+                    try:
+                        pods = json.loads(value)
+                    except ValueError:
+                        pass
+                #
+                # - if the wildcard character is used, we need to watch: 
+                #   1) the ROOT node in case new children matching the desired
+                #      tag are added/deleted
+                #   2) any existing child matching the desired tag as those will not
+                #      trigger a notification on the parent node in case they are modified
+                #
+                else:
+                    path = ROOT
+                    childRegex = re.escape(self.scope + '.') + self.tag.replace("*", ".*")
+                    for child in self.zk.get_children(path, watch=self.feedback):
+                        if re.search(childRegex, child):
+                            fullPath = '%s/%s/snapshot' % (ROOT, child)
+                            # 
+                            # - the parent node may exist before the 'snapshot' node does
+                            #
+                            if self.zk.exists(fullPath, watch=self.feedback):                            
+                                try:
+                                    value, stat = self.zk.get(fullPath, watch=self.feedback)
+                                    pods = {k: v for (k, v) in (json.loads(value).items() + pods.items())}
```

Comment:
i believe the pods dict will get overriden in that for loop .. shouldn't it be a concatenation instead of an assignment ?